File: app/src/main/java/com/fanwe/log/MainActivity.java
package com.fanwe.log;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.fanwe.lib.log.FLogger;

public class MainActivity extends AppCompatActivity
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FLogger.get(AppLogger.class).info("onCreate");
    }

    @Override
    protected void onResume()
    {
        super.onResume();
        FLogger.get(AppLogger.class).info("onResume");
    }

    @Override
    protected void onStop()
    {
        super.onStop();
        FLogger.get(AppLogger.class).info("onStop");
    }

    @Override
    protected void onDestroy()
    {
        super.onDestroy();
        FLogger.get(AppLogger.class).info("onDestroy");
    }
}

File: app/src/main/java/com/sd/demo/log/MainActivity.kt
package com.sd.demo.log

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.sd.demo.log.databinding.ActivityMainBinding
import com.sd.lib.log.FLogger
import com.sd.lib.log.flogI
import kotlin.time.measureTime

class MainActivity : AppCompatActivity() {
    private val _binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(_binding.root)
        _binding.btn.setOnClickListener {
            // 删除日志文件
            FLogger.deleteLogFile()
        }
    }

    override fun onStart() {
        super.onStart()
        flogI<AppLogger> { "onStart" }
    }

    override fun onStop() {
        super.onStop()
        flogI<AppLogger> { "onStop" }
    }
}

/**
 * 日志性能测试，请在关闭控制台日志的情况下测试[FLogger.open]
 */
private fun testLogPerformance(times: Int = 1_0000) {
    val log = "1".repeat(512)
    measureTime {
        repeat(times) {
            flogI<AppLogger> { log }
        }
    }.let {
        Log.i(MainActivity::class.java.simpleName, "time:${it.inWholeMilliseconds}")
    }
}